huaweicloud/config.go

```diff
@@ -416,16 +426,39 @@ func (c *Config) NewServiceClient(srv, region string) (*golangsdk.ServiceClient,
 	if allServiceCatalog[srv].Admin {
 		client = c.DomainClient
 	}
-	return c.newServiceClientByName(client, allServiceCatalog[srv], c.Region)
+	return c.newServiceClientByName(client, allServiceCatalog[srv], region)
 }
 
 func (c *Config) newServiceClientByName(client *golangsdk.ProviderClient, catalog ServiceCatalog, region string) (*golangsdk.ServiceClient, error) {
 	if catalog.Name == "" || catalog.Version == "" {
 		return nil, fmt.Errorf("must specify the service name and api version")
 	}
 
+	// Resource-level region only supports AK/SK authentication.
+	if region != c.Region && (c.AccessKey == "" || c.SecretKey == "") {
+		return nil, fmt.Errorf("Resource-level region must be same as Provider-level region when not AK/SK authentication if Resource-level region setted")
```

Comment:
s/setted/set

Reply:
"Resource-level region must be same as Provider-level region when using non AK/SK authentication" seems better :P

Reply:
Cool, that seems better truly.